### PR TITLE
OPS-2366, Issue #919 Add merge-keys feature

### PR DESCRIPTION
**Pull request by alex-harvey-z3q**
Adds CLI option `--merge-keys`. Dictionaries in multiple var files are
merged instead of overwritten. Also dicts specified via `--var` are
merged instead of overwritten. Adds test cases.

### CODE REVIEW

File: sceptre/cli/__init__.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # -*- coding: utf-8 -*-
 
+# flake8: noqa
```
Comment (mrowlingfox) [inline sceptre/cli/__init__.py:3]: Why's this required?

Comment (alex-harvey-z3q) [reply sceptre/cli/__init__.py:3]: Without it, flake8 fails the tests with C019 function too complex. I could not find any way to disable that specific check or that specific function.

Comment (alex-harvey-z3q) [reply sceptre/cli/__init__.py:3]: Actually I could avoid falling foul of the complexity check by defining my function `deep_merge` outside of the `cli` function. I don't think that leads to improvement in the code however but it is an option.

Comment (mrowlingfox) [reply sceptre/cli/__init__.py:3]: It would be good to move it out into some sort of utilities file. Makes it easier to test too

**Review by mrowlingfox — DISMISSED** [review]
Can you please remove the `j2_extension` behaviour Then LG2M

**Review by mrowlingfox — APPROVED** [review]

**Pull request merged**